genie-web/src/main/java/com/netflix/genie/web/agent/launchers/impl/TitusAgentLauncherImpl.java

```diff
@@ -339,14 +340,39 @@ private TitusBatchJobRequest createJobRequest(final ResolvedJob resolvedJob) thr
                     .detail(this.titusAgentLauncherProperties.getDetail())
                     .sequence(this.titusAgentLauncherProperties.getSequence())
                     .build()
-            )
-            .build();
+            );
+
+        final Optional<String> networkConfiguration = validateNetworkConfiguration(this.environment.getProperty(
+            TitusAgentLauncherProperties.CONTAINER_NETWORK_CONFIGURATION,
+            String.class));
+        networkConfiguration.ifPresent(config -> requestBuilder.networkConfiguration(
+                TitusBatchJobRequest.NetworkConfiguration.builder().networkMode(config).build()));
+
+        final TitusBatchJobRequest request = requestBuilder.build();
 
         // Run the request through the security adapter to add any necessary context
         this.jobRequestAdapter.modifyJobRequest(request, resolvedJob);
         return request;
     }
 
+    private Optional<String> validateNetworkConfiguration(@Nullable final String networkConfig) {
+        if (Strings.isNullOrEmpty(networkConfig)) {
+            return Optional.empty();
+        }
+
+        switch (networkConfig) {
+            case "Ipv4Only":
+            case "Ipv6AndIpv4":
+            case "Ipv6AndIpv4Fallback":
+            case "Ipv6Only":
+            case "HighScale":
```

Comment:
can we make these constants, and at the constants definition, link to where this is from (https://github.com/Netflix/titus-api-definitions/blob/master/doc/titus-v3-spec.md#networkconfigurationnetworkmode) ?